### PR TITLE
[lldb] Document SymbolFileJSON

### PR DESCRIPTION
**Pull request by JDevlieghere**
I've had multiple request for documentation about the JSON symbol file format that LLDB supports. This patch documents the structure and fields, shows a handful of examples and explains how to use it in LLDB.

### CODE REVIEW

**Comment by medismailben** [comment]
LGTM but it would be nice if we made the `optional` fields stand out more by either making it **bold**  in the paragraph or add an `(optional)` on the example lists.

**Review by medismailben — APPROVED** [review]
Also may be you can point to the crashlog script as an example where this is already being used.

File: lldb/docs/resources/symbolfilejson.rst [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+=======================
+
+The JSON symbol file format encodes symbols in a text based, human readable
+format. JSON symbol files can be used to symbolicate programs that miss symbol
+information, for example because they have been stripped.
+
+Format
+------
+
+The symbol file contains of a single JSON object with the following top level
```
Comment (bulbazord) [inline lldb/docs/resources/symbolfilejson.rst:11]: `contains of a` -> `contains a`

File: lldb/docs/resources/symbolfilejson.rst [hunk 2]
```diff
@@ -0,0 +1,201 @@
+JSON Symbol File Format
+=======================
+
+The JSON symbol file format encodes symbols in a text based, human readable
+format. JSON symbol files can be used to symbolicate programs that miss symbol
```
Comment (bulbazord) [inline lldb/docs/resources/symbolfilejson.rst:5]: Suggestion: `miss` -> `lack`

File: lldb/docs/resources/symbolfilejson.rst [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+The optional ``type`` field allows you to specify the type of object file the
+JSON file represent. This is often unnecessary, and can be omitted, in which
+case the file is considered of the type ``DebugInfo``.
+
+Valid values for the ``type`` field are:
+
+* ``corefile``: A core file that has a checkpoint of a program's execution state.
+* ``executable``: A normal executable.
+* ``debuginfo``: An object file that contains only debug information.
+* ``dynamicuinker``: The platform's dynamic linker executable.
```
Comment (bulbazord) [inline lldb/docs/resources/symbolfilejson.rst:60]: `dynamicuinker` -> `dynamiclinker`

**Review by bulbazord — APPROVED** [review]

**Pull request merged**